server/tpm2.go

```diff
@@ -4,10 +4,125 @@
+
 	"github.com/google/go-tpm/tpm2"
 	"github.com/google/go-tpm/tpmutil"
+	"github.com/sirupsen/logrus"
 )
 
+// TCG TPM v2.0 Provisioning Guidance, section 7.8:
+// https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-v2.0-Provisioning-Guidance-Published-v1r1.pdf
+const SRK_HANDLE tpmutil.Handle = 0x81000001
+
+// TCG TPM v2.0 Provisioning Guidance, section 7.5.1:
+// https://trustedcomputinggroup.org/wp-content/uploads/TCG-TPM-v2.0-Provisioning-Guidance-Published-v1r1.pdf
+var SRK_TEMPLATE = tpm2.Public {
+	Type: tpm2.AlgRSA,
+	NameAlg: tpm2.AlgSHA256,
+	Attributes: tpm2.FlagFixedTPM | tpm2.FlagFixedParent | tpm2.FlagSensitiveDataOrigin | tpm2.FlagUserWithAuth | tpm2.FlagRestricted | tpm2.FlagDecrypt | tpm2.FlagNoDA,
+	AuthPolicy: nil,
+	RSAParameters: &tpm2.RSAParams {
+		Symmetric: &tpm2.SymScheme {
+			Alg: tpm2.AlgAES,
+			KeyBits: 128,
+			Mode: tpm2.AlgCFB,
+		},
+		KeyBits: 2048,
+		ModulusRaw: make([]byte, 256),
+	},
+}
+
+/*
+	Returns a handle to the TPM Storage Rook Key (SRK) if it's already present
+	in it's non volatile memory.
+	Else, a new SRK is created using the default template, and is persisted
+	in the TPM Non Volatile memory, at it's default location. A handle to the newly
+	created key is then returned
+*/
+func TPM2_LoadSRK(rwc io.ReadWriteCloser) (tpmutil.Handle, error) {
+	handles, _, err := tpm2.GetCapability(rwc, tpm2.CapabilityHandles, 1000, uint32(tpm2.HandleTypePersistent) << 24)
+	if err != nil {
+		return 0, nil
+	}
+	for _, h := range handles {
+		if h.(tpmutil.Handle) == SRK_HANDLE {
+			return SRK_HANDLE, nil
+		}
+	}
+	logrus.Info("SRK not found, creating one. This may take a while.")
+
+	handle, _, err := tpm2.CreatePrimary(rwc, tpm2.HandleOwner, tpm2.PCRSelection{}, "", "", SRK_TEMPLATE)
+	if err != nil {
+		return 0, err
+	}
+	if err = tpm2.EvictControl(rwc, "", tpm2.HandleOwner, handle, SRK_HANDLE); err != nil {
+		return 0, err
+	}
+	logrus.Info("Persistent SRK created at NV index %x\n", SRK_HANDLE)
+	return SRK_HANDLE, nil
+}
+
+/*
+	Seals the given data to the TPM Storage Root Key (SRK) and
+	returns the resulting private and public blobs
+*/
+func TPM2_Seal(data []byte) ([]byte, []byte, error) {
+	var rwc io.ReadWriteCloser
+	var priv, pub, policy []byte
+	var srkHandle, sessHandle tpmutil.Handle
+	var err error
+
+	if rwc, err = tpm2.OpenTPM(); err != nil {
+		return nil, nil, err
+	}
+	defer rwc.Close()
+
+	if srkHandle, err = TPM2_LoadSRK(rwc); err != nil {
+		return nil, nil, err
+	}
+	if sessHandle, _, err = tpm2.StartAuthSession(rwc, tpm2.HandleNull, tpm2.HandleNull, make([]byte, 16), nil, tpm2.SessionPolicy, tpm2.AlgNull, tpm2.AlgSHA256); err != nil {
+		return nil, nil, err
+	}
+	if policy, err = tpm2.PolicyGetDigest(rwc, sessHandle); err != nil {
+		return nil, nil, err
+	}
+	if priv, pub, err = tpm2.Seal(rwc, srkHandle, "", "", policy, data); err != nil {
+		return nil, nil, err
+	}
+	return priv, pub, err
+}
+
+/*
+	Tries to unseal the given object with the TPM Storage Root Key (SRK)
```

Comment:
Remove "tries to" : every function can fail, this is already indicated by `error` in the return type.